x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -279,7 +279,8 @@ protected LogicalPlan rule(Aggregate agg) {
                 }
             }
 
-            if (prunedGroupings.size() > 0) {
+            // avoid complete removal of the grouping, that could change the query semantics
```

Comment:
```suggestion
            // avoid complete removal of the grouping because it would turn the aggregation into an implicit grouping and change the semantics
```

I think the wording can be a bit stronger here because, as far as I understand, it always changes the semantics.

Reply:
In that case there wouldn't be any groupings to remove though?

Reply:
In that case the explicit groupings (constants), if present, can be safely removed.